src/app/components/playlist_details/playlist_details.rs

```diff
@@ -57,8 +59,18 @@ impl PlaylistDetails {
     fn update_details(&self) {
         if let Some(info) = self.model.get_playlist_info() {
             let title = &info.title[..];
+            let owner = &info.owner.display_name[..];
 
             self.widget.name_label.set_label(title);
+            self.widget.owner_button_label.set_label(owner);
+
+            let weak_model = Rc::downgrade(&self.model);
```

Comment:
can't blame, probably inspired by existing code, but you can use [clone!](https://gtk-rs.org/docs/glib/macro.clone.html#passing-a-weak-reference) to avoid manually downgrading model

Reply:
Ah, yes forgot about that macro

Reply:
Should be fine, unless I'm missing something! This update fn is only called for a specific event, `AppEvent::BrowserEvent(BrowserEvent::AlbumDetailsLoaded(id))`, so it should not be called more than once. Maybe the naming (update_xxx) is confusing and should be changed?

Reply:
probably fine :). Is that not called every time a new album is displayed?